lib/gnu/md5.c

```diff
@@ -299,13 +299,10 @@ md5_process_bytes (const void *buffer, size_t len, struct md5_ctx *ctx)
       if (left_over >= 64)
         {
           md5_process_block (ctx->buffer, 64, ctx);
-          left_over &= 63; /* same as -= 64 but easier to understand
-                              for static analysis
-                            */
+          left_over -= 64;
           /* The regions in the following copy operation cannot overlap,
              because left_over ≤ 64.  */
-          memcpy (ctx->buffer, &(((char *) ctx->buffer)[64]), left_over);
-          /* cast to char * as hint for static analysis */
+          memcpy (ctx->buffer, &ctx->buffer[16], left_over);
```

Comment:
These were local changes by @wenningerk in 2ba4cc6, for the last time release testing had to be done. :) I guess we could make a note of that somewhere. Either that or we should ignore static analysis for md5.c and not touch it.

Reply:
I'm happy with just ignoring it.  Whatever we do here will just get stepped on when we grab an updated gnulib.

Reply:
Don't know if it is still useful with more current versions of tooling. Either more picky or more intelligent ...
What we could do in theory is adding a patch to the gnulib-import-script. Shouldn't be too risky that it does something unwanted silently as the patch wouldn't apply anymore. Haven't played with coccinelle myself to be able to tell if things like that would make sense to be tackled with it in a more generic way.
I just made it more similar to code a bit above that tooling obviously didn't complain about.

Reply:
Bah, I see what happened here.  I'll revert this portion and we can figure out what to do about gnulib in general another day.